### PR TITLE
Verify docker image digest before opting in for docker image cache feature [BT-94]

### PR DESCRIPTION
**Pull request by grsterin**

### CODE REVIEW

**Review by kshakir — APPROVED** [review]
I'm happy when codecov is happy:

```
5.55% of diff hit (target 50.00%)
```

File: src/ci/resources/papi_v2beta_application.conf [hunk 1]
```diff
@@ -89,7 +89,7 @@ backend {
         include "papi_v2beta_provider_config.inc.conf"
         genomics.compute-service-account = "centaur@broad-dsde-cromwell-dev.iam.gserviceaccount.com"
         filesystems.http {}
-        docker-image-cache-manifest-file = "gs://cloud-cromwell-dev/docker-image-cache-manifest.json"
+        docker-image-cache-manifest-file = "gs://cloud-cromwell-dev/docker-image-cache-manifest-v2.json"
```
Comment (mcovarr) [inline src/ci/resources/papi_v2beta_application.conf:92]: TOL ISO 8601 embedded date formats might be nice, e.g. `docker-image-cache-manifest-2021-01-21.json`

File: supportedBackends/google/pipelines/common/src/test/resources/docker-image-cache-manifest.json [hunk 2]
```diff
@@ -1,4 +1,10 @@
...
-  "project1/dockerImage1": "projects/some-google-project/global/images/dockerCacheDiskForDockerImage1",
-  "project2/dockerImage2": "projects/another-google-project/global/images/dockerCacheDiskForDockerImage2"
+  "project1/dockerImage1": {
+    "dockerImageDigest": "sha256:c73e11b1a7854f31ff12c607738bef7b0560a880dc9d0445dde084acc0e9da09",
+    "diskImageName": "projects/some-google-project/global/images/dockerCacheDiskForDockerImage1"
+  },
+  "project2/dockerImage2": {
+    "dockerImageDigest": "sha256:7eb386481d87e41ebddceb948f25379bb339784df5247a3dfbdea2ac101b10c0",
+    "diskImageName": "projects/another-google-project/global/images/dockerCacheDiskForDockerImage2"
+  }
```
Comment (mcovarr) [inline supportedBackends/google/pipelines/common/src/test/resources/docker-image-cache-manifest.json:9]: Inspired by these changes, perhaps there should be a `version` field in this manifest format as well. I do like `version 2` for this as it is a breaking change from the implicit `version 1`.

**Review by mcovarr — APPROVED** [review]
TOL should this code check if the manifest version is a supported one?

**Pull request merged**